tests/Fixer/Basic/NumericLiteralSeparatorFixerTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Basic;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Marvin Heilemann <11534760+muuvmuuv@users.noreply.github.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Basic\NumericLiteralSeparatorFixer
+ */
+final class NumericLiteralSeparatorFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public static function provideFixCases(): iterable
+    {
+        $cases = [
+            'decimal' => [
+                '1234' => '1_234',
+                '-1234' => '-1_234',
+                '12345' => '12_345',
+                '123456' => '123_456',
+                '1_2_3_4' => '1_234',
+            ],
+            'binary' => [
+                '0b0101010001101000' => '0b0101_0100_0110_1000',
+                '0B01010100_01101000' => '0B0101_0100_0110_1000',
+                '0b01010100011010000110010101101111' => '0b0101_0100_0110_1000_0110_0101_0110_1111',
+                '0b110001000' => '0b1_1000_1000',
+            ],
+            'float' => [
+                '1234.5' => '1_234.5',
+                '1.2345' => '1.234_5',
+                '01234.5' => '01_234.5',
+                '1234e5' => '1_234e5',
+                '1234E5' => '1_234E5',
+                '1e2345' => '1e2_345',
+                '1234.5678e1234' => '1_234.567_8e1_234',
+                '1.1e-1234' => '1.1e-1_234',
+                '1.1e-12345' => '1.1e-12_345',
+                '1.1e-123456' => '1.1e-123_456',
+                '01234e5' => '01_234e5',
+            ],
+            'hexadecimal' => [
+                '0x42726F776E' => '0x42_72_6F_77_6E',
+                '0X42726F776E' => '0X42_72_6F_77_6E',
+                '0x2726F776E' => '0x2_72_6F_77_6E',
+                '0x1234567890abcdef' => '0x12_34_56_78_90_ab_cd_ef',
+                '0X1234567890ABCDEF' => '0X12_34_56_78_90_AB_CD_EF',
+                '0x1234e5' => '0x12_34_e5',
+            ],
+            'octal' => [
+                '012345' => '012_345',
+                '0123456' => '0_123_456',
+                '01234567' => '01_234_567',
+            ],
```

Comment:
quesiton

I wonder if notation for hexal and octal is normalized somehow, ie if we should put `0x_` or `0_` prefix , or sth?
for format you applied, where is it coming from, if any?
ie, `012_345` vs `0_123_456` looks counterintuitive for me a bit (i know you divided into max 3 digits per section, but that;s now what we done for `0x...`)

Reply:
There is this RFC where he added examples which I tried to match https://wiki.php.net/rfc/numeric_literal_separator

Initially, I had this pattern, i.e. remove 0 prefix, but I decided not to since this fixer just adds the separator, not correct wrong number formations. But let's discuss.

The thing with Octal is that it's hard to determine if it's an int or oct, there is no const for it (at least couldn't find any). So, I just fall back to int handling.

For hex, I often see that it's separated by 2 (hex edit etc.), but I am unaware of any official rules. Same for binary. We could make this configurable? PHP does not care at which position the underscore appears, it seems to be just visual.